drivers/modem/modem_cellular.c

```diff
@@ -1861,3 +1950,61 @@ DT_INST_FOREACH_STATUS_OKAY(MODEM_CELLULAR_DEVICE_SWIR_HL7800)
 #define DT_DRV_COMPAT telit_me910g1
 DT_INST_FOREACH_STATUS_OKAY(MODEM_CELLULAR_DEVICE_TELIT_ME910G1)
 #undef DT_DRV_COMPAT
+
+int cellular_get_modem_info(const struct device *dev, enum cellular_modem_info_type type,
+			    char *info, size_t size)
+{
+	int rc = 0;
+
+	struct modem_cellular_data *data = (struct modem_cellular_data *)dev->data;
+
+	switch (type) {
+	case CELLULAR_MODEM_INFO_IMEI:
+		memcpy(info, &data->imei[0], MIN(size, sizeof(data->imei)));
+		break;
+	case CELLULAR_MODEM_INFO_SIM_IMSI:
+		memcpy(info, &data->imsi[0], MIN(size, sizeof(data->imsi)));
+		break;
+	case CELLULAR_MODEM_INFO_SIM_ICCID:
+		memcpy(info, &data->iccid[0], MIN(size, sizeof(data->iccid)));
+		break;
+	case CELLULAR_MODEM_INFO_MANUFACTURER:
+		memcpy(info, &data->manufacturer[0], MIN(size, sizeof(data->manufacturer)));
+		break;
+	case CELLULAR_MODEM_INFO_FW_VERSION:
+		memcpy(info, &data->fw_version[0], MIN(size, sizeof(data->fw_version)));
+		break;
+	case CELLULAR_MODEM_INFO_MODEL_ID:
+		memcpy(info, &data->model_id[0], MIN(size, sizeof(data->model_id)));
+		break;
+	default:
+		rc = -ENODATA;
+		break;
+	}
+
+	return rc;
+}
```

Comment:
We are working with char strings here, so we need to NULL terminate the char array :), use `snprintk()` or `strncpy` instead of `memcpy`